language/stdlib/modules/Diem.move

```diff
@@ -500,6 +518,35 @@ module Diem {
 
     }
 
+
+
+    ///////////////////////////////////////////////////////////////////////////
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Upgrade to Concurrent Preburns for DDs
+    ///////////////////////////////////////////////////////////////////////////
+
+    // when dd tries to preburn
+    public fun remove_preburn_singleton_resource_if_exists<CoinType>(account: &signer) acquires Preburn<CoinType> {
+        let sender = Signer::address_of(account);
+        if (exists<Preburn<CoinType>>(sender)) {
+            let Preburn<CoinType> {} = move_from<Preburn<CoinType>>(sender);
+        }
+    }
+
+    // when DD tries to preburn ideally but actually need tc signer..to create preburn thing
+    public fun create_concurrent_preburn_resource<CoinType>(
```

Comment:
nonpub

Reply:
do in same function (only be published by TC or DD which already had preburn) 
then need another function for new create DD flow which has a tc account signer.